### ISSUE TITLE
craftpoker.com isn't remembering nicks and passwords (at least on brave)

### ISSUE THREAD

**Issue by ctm** [ctm/mb2-doc#1145]
Fix it so craftpoker.com works like devctm.com w.r.t. remembering passwords.

This one is puzzling. I can't think of any difference that would cause this, but I haven't yet done any web-searching or poking around.

> JADC🚴🏼: The old devctm login page held my bike emoji and my password. The craftpoker one does not. The password is no big deal, but I'm not smart enough to figure out how to save that bike emoji.
deadhead: I'm noticing weirdness with Brave not remembering my nick and password on craftpoker.com, too.
deadhead: I'll make a bug report and figure out the nick/password issue, JADC.

**Comment by ctm** [comment]
The problem exists on Chrome and Brave, but not Safari or Firefox. I've fiddled around a bit and done some web searching and so far, the only thing I can tell is that Chrome is not putting up the little key icon when interacting with the login page, so for some reason it's deciding password saving isn't appropriate.

**Comment by ctm** [comment]
If I use Brave's password manager's Add button, I can add (craftpoker.com, deadhead, mypassword) and then when I get to the login page, if I focus on the username field, the option to autofill appears and the little key icon appears. If I then delete that info, I'm back to where I was, in that there's no offer to save the password and no little key icon.

**Comment by ctm** [comment]
I tried creating a new user on my development machine and that new user had the same problem in Chrome. My guess is Chrome is unhappy with some aspect the SSL certificate just enough to prevent it from recognizing the password field as something that should get the password manager's offer, but not enough to show up as a warning anywhere or to otherwise interfere with the site. 

The above would seem like an amazingly bad thing for Chrome to do, and if that were the case, I would assume I could find such info by doing a web search, but previously devctm.com was the primary certificate name (CN) and craftpoker.com was an alternate. I've now generated a certificate that has craftpoker.com as the primary with devctm.com as the alternate, but even after a redeploy, when I check the craftpoker.com certificate, I'm seeing one that was issued by Google and not Let's Encrypt, so I have to figure out what that's about.

**Comment by ctm** [comment]
The Google certificate is a Cloudflare edge certificate. Craftpoker.com's Cloudflare edge certificate lasts for three months and is issued by Google. Devctm.com's Cloudflare edge certificate lasts for a year and is issued by Cloudflare. I'm not sure what's going on there. My guess is that Cloudflare simply changed what they're doing.

Since I was poking around on Cloudflare anyway, I realized there were some DNS settings that differed, for example, craftpoker.com had no MX record. As such, I made new craftpoker.com DNS records for almost all the records that devctm.com has, hoping that would change things.  It didn't.

I briefly rolled back the craftpoker.com <-> devctm.com swap so mb2's login page was back on devctm.com just to make sure that if I did that, Chrome's password manager would work again and sure enough, it did.

I'm now wondering if there's some petty blacklist that disables the password manager if "poker" is part of the domain name. That seems unlikely, but I'm grasping at straws now.

**Comment by ctm** [comment]
FWIW, I found the [Cloudflare documentation](https://developers.cloudflare.com/ssl/reference/certificate-authorities/) that says they've switched from DigiCert to Google Trust Services and backed off from one year certificates to three month ones. I'm now seeing if I can nuke the devctm.com Cloudflare edge certificate and get a new one issued, because the new one would be Google Trust Services, and that would allow me to know if that difference is causing the problem. It seems **extremely** unlikely to me that Chrome would be unhappy with GTS services, because presumably tons of companies are using them, but I do like to eliminate differences when trying to figure out what changed.

**Comment by ctm** [comment]
I did find a way to get a new (i.e., GTS) edge certificate for devctm.com. I'm now, once again, rolling back the craftpoker.com <-> devctm.com swap, to see if the new certificate causes problems when devctm.com has the login page.

**Comment by ctm** [comment]
Ugh. My deploy blew up both devctm.com and craftpoker.com, and I'm at a location with a slow internet upload. So, although I've corrected the problem and started a redeploy, it'll be another eight minutes or so before the sites are back.

**Comment by ctm** [comment]
I made the minimal change so that the exact same HTML was being served from craftpoker.com as was being served from devctm.com and the problem persisted.

A blacklist that includes the word poker seems very unlikely to me, but still seems possible. I'll ask the BARGE list if anyone has any ideas.

**Comment by ctm** [comment]
I found a way to view the Chrome error logs and errors are indeed being reported.  The trick is to open chrome://autofill-internals in a tab an then go to craftpoker.com. Now I need to read through the complaints and figure them out.

Whee!

**Comment by ctm** [comment]
For whatever reason (and I strongly suspect a bug in Chrome), Chrome was thinking some fields on hidden forms on craftpoker.com  were visible and focusable, but not on devctm.com. Those forms didn't even need to be generated for the login page, however, since they were invisible, I never noticed that mb2 was generating them for the login page. So, I made it so that they're no longer generated for the login page and that fixed the problem.

If this _is_ a bug in Chrome, it would be worthwhile to report it, but I don't have time to the triage needed to figure that out. Perhaps after the WSOPS series is successfully running (and I've added late registration).

[devctm-autofill-internals.pdf](https://github.com/ctm/mb2-doc/files/11568334/devctm-autofill-internals.pdf)
[craftpoker-autofill-internals.pdf](https://github.com/ctm/mb2-doc/files/11568335/craftpoker-autofill-internals.pdf)
